util/efd_sched.py

```diff
@@ -0,0 +1,231 @@
+"""Developer utilities to query the EFD for schedulers, and download them.
+"""
+
+import argparse
+import asyncio
+import lzma
+import os
+import pickle
+import re
+
+import astropy.units as u
+from astropy.time import Time
+from lsst.resources import ResourcePath
+from lsst_efd_client import EfdClient
+from rubin_sim.utils import Site
+
+
+async def query_schedulers_in_window(desired_time, efd="usdf_efd", time_window_seconds=2):
+    """Query the EFD for scheduler URLs within a given time window.
+
+    Parameters
+    ----------
+    desired_time : `str`
+        The central time to query the event service in ISO format
+        (YYYY-MM-DDTHH:MM:SS).
+    efd : `str`, optional
+        The name of the EFD to connect to (default is 'usdf_efd').
+    time_window_seconds : `float`, optional
+        The size of the time window to search for scheduler URLs, in seconds
+        (default is 2).
+
+    Returns
+    -------
+    scheduler_urls : `pandas.DataFrame`
+        A DataFrame containing the scheduler snapshot times URLs.
+    """
+    desired_time = Time(desired_time)
+    start_time = desired_time - (time_window_seconds / 2) * u.second
+    end_time = desired_time + (time_window_seconds / 2) * u.second
+
+    efd_client = EfdClient(efd)
+    topic = "lsst.sal.Scheduler.logevent_largeFileObjectAvailable"
+    fields = ["url"]
+    scheduler_urls = await efd_client.select_time_series(topic, fields, start_time, end_time)
```

Comment:
It likely doesn't matter at the moment, but I believe the salIndex = 2 means "for aux tel" while 1 would be main tel. 
Without definition, I don't know what you get (at the moment only auxtel is active, so we just get auxtel).
Probably we just need to add a kwarg so that we can specify which to retrieve, as the time is coming not too far in the future where we'd want main tel things too.

Reply:
I don't expect this code to live long enough for anything other that auxtel to be relevant. There's no harm in supporting the main telescope as well, though.